File: ruby/resistor-color-duo/resistor_color_duo.rb
# Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
# `resistor_color_duo_test.rb` pass.
#
# To get started with TDD, see the `README.md` file in your
# `ruby/resistor-color-duo` directory.

module ResistorColorDuo
  COLORS = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }.freeze

  def self.value(color)
    color.take(2).each_with_index.reduce(0) do |acc, (c, i)|
      acc + COLORS[c.to_sym] * 10**(1 - i)
    end
  end
end

File: go/resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

var encoded = map[string]int{
	"black":  0,
	"brown":  1,
	"red":    2,
	"orange": 3,
	"yellow": 4,
	"green":  5,
	"blue":   6,
	"violet": 7,
	"grey":   8,
	"white":  9,
}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	return encoded[colors[0]]*10 + encoded[colors[1]]
}
